File: topicos/dynamic-programming/group-1/kadane.ts
export function kadake(nums: number[]) {
    if (!nums.length) return 0;


    let currMax = nums[0];
    let globalMax = nums[0];

    for (let i = 1; i < nums.length; i++) {
        currMax = Math.max(nums[i], currMax+nums[i])
        
        if (currMax > globalMax) {
            globalMax = currMax;
        }
    }

    return globalMax;
}

File: topicos/dynamic-programming/group-1/kadane/kadane.go
package main

func kadane(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	currMax := nums[0]
	globalMax := nums[0]

	for int i := 1; i < len(nums); i++ {
		if nums[i] > currMax+nums[i] {
			currMax = nums[i]
		} else {
			currMax += nums[i]
		}

		if currMax < globalMax {
			globalMax = currMax
		}
	}

	return globalMax
}